include/zephyr/net/dhcpv4.h

```diff
@@ -163,6 +169,48 @@ int net_dhcpv4_remove_option_callback(struct net_dhcpv4_option_callback *cb);
 
 #endif /* CONFIG_NET_DHCPV4_OPTION_CALLBACKS */
 
+#ifdef CONFIG_NET_DHCPV4_OPTION_CALLBACKS_VENDOR_SPECIFIC
+
+/**
+ * @brief Helper to initialize a struct net_dhcpv4_option_callback for encapsulated vendor-specific
+ * options properly
+ * @param callback A valid Application's callback structure pointer.
+ * @param handler A valid handler function pointer.
+ * @param option The DHCP encapsulated vendor-specific option the callback responds to.
+ * @param data A pointer to a buffer for max_length bytes.
+ * @param max_length The maximum length of the data returned.
+ */
+static inline void
+net_dhcpv4_init_option_vendor_callback(struct net_dhcpv4_option_callback *callback,
+				       net_dhcpv4_option_callback_handler_t handler, uint8_t option,
+				       void *data, size_t max_length)
+{
+	__ASSERT(callback, "Callback pointer should not be NULL");
+	__ASSERT(handler, "Callback handler pointer should not be NULL");
+	__ASSERT(data, "Data pointer should not be NULL");
+
+	callback->handler = handler;
+	callback->option = option;
+	callback->data = data;
+	callback->max_length = max_length;
```

Comment:
As this is application interfacing API, I would suggest that we check the parameters properly and return error codes etc.

Reply:
@jukkar Since I had asked Fin to base this code on my option callback code (#57918), this part, as well as the one mentioned in your slist comment are pretty much exact copies of the code for regular callbacks.
Since your comments also apply to that already mearged code, should we redo those as well? If so, should that be a separate PR or just part of this one?

Reply:
Ah, I missed that the existing function was doing the same thing. Hmm, lets leave this like this for now and deal with this later then.

Reply:
Probably for the better, since my code in turn was based on the gpio callback API (https://github.com/zephyrproject-rtos/zephyr/blob/40b53d5d774705d27183aaa1e42f2f322250da10/include/zephyr/drivers/gpio.h#L1692)..